inference/core/workflows/core_steps/models/roboflow/instance_segmentation.py

```diff
@@ -200,10 +198,14 @@ async def run_locally(
         disable_active_learning: Optional[bool],
         active_learning_target_dataset: Optional[str],
     ) -> List[Dict[str, Union[sv.Detections, Any]]]:
+        non_empty_images = [i for i in images.iter_nonempty()]
+        non_empty_inference_images = [
+            i.to_inference_format(numpy_preferred=True) for i in non_empty_images
+        ]
```

Comment:
this pattern seems to be repeating across blocks, maybe `images.iter_nonempty` can take `numpy_preferred` as parameter and return required inference images in one go?

Reply:
👎